plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1398,6 +1398,22 @@ public Optional<ConnectorOutputMetadata> finishInsert(
             transactionLogWriter.flush();
             writeCommitted = true;
             writeCheckpointIfNeeded(session, new SchemaTableName(handle.getSchemaName(), handle.getTableName()), checkpointInterval, commitVersion);
+
+            if (isCollectExtendedStatisticsColumnStatisticsOnWrite(session) && !computedStatistics.isEmpty() && !dataFileInfos.isEmpty()) {
+                // TODO (https://github.com/trinodb/trino/issues/16088) Add synchronization when version conflict for INSERT is resolved.
+                Optional<Set<String>> analyzedColumns = statisticsAccess.readExtendedStatistics(session, handle.getLocation())
+                        .flatMap(ExtendedStatistics::getAnalyzedColumns);
+                Optional<Instant> maxFileModificationTime = dataFileInfos.stream()
+                        .map(DataFileInfo::getCreationTime)
+                        .max(Long::compare)
+                        .map(Instant::ofEpochMilli);
+                updateTableStatistics(
```

Comment:
The target column names may contain uppercase characters when the table was created by other query engine. Don't we need to preserve original cases?

Reply:
Delta lake internally uses always lower case column names, even when underlying column names have uppercase characters. So the extended statistics are also stored with lower case column names.
This case is covered by test cases:
[testInsertIntoNonLowercaseColumnTable](https://github.com/trinodb/trino/pull/16026/files#diff-5053d5f66df3044008495f77d7722a7785aef8bd22c26e5b4f95819cb82c31a4L919)
[testInsertNestedNonLowercaseColumns](https://github.com/trinodb/trino/pull/16026/files#diff-5053d5f66df3044008495f77d7722a7785aef8bd22c26e5b4f95819cb82c31a4L955)
[testInsertIntoPartitionedNonLowercaseColumnTable](https://github.com/trinodb/trino/pull/16026/files#diff-5053d5f66df3044008495f77d7722a7785aef8bd22c26e5b4f95819cb82c31a4R1017)

Reply:
> Delta lake internally uses always lower case column names

Not exactly. See https://github.com/trinodb/trino/pull/16026/files#diff-79f367e47040dda32ac81ecec46b27cfd7cc822858bc8c31fe435a15a47446dcR1394. I'm fine with the current implementation though.